projects/scripts/adi_board.tcl

```diff
@@ -617,79 +617,91 @@ proc ad_mem_hpc1_interconnect {p_clk p_name} {
   if {$sys_zynq == 2} {ad_mem_hpx_interconnect "HPC1" $p_clk $p_name}
 }
 
-## Create an memory mapped interface connection to a MIG or PS7/8 IP, using a
-#  HP0 high speed interface in case of PSx.
+## Create a memory mapped interface connection to a MIG or PS7/8 IP, using an
+#  HP0 high speed interface in case of PSx. On UltraScale, if cache coherency
+#  is enabled, an HPC0 high performance coherent interface will be used instead.
 #
 #  \param[p_clk]  - name of the clock or reset source
 #  \param[p_name] - name or list of names of the clock or reset sink
+#  \param[p_cc]   - cache coherency support
 #
-proc ad_mem_hp0_interconnect {p_clk p_name} {
+proc ad_mem_hp0_interconnect {p_clk p_name {p_cc false}} {
 
   global sys_zynq
+  global sys_hpc0_interconnect_index
 
-  if {($sys_zynq != 1 && $sys_zynq != 2) && ($p_name eq "sys_ps7/S_AXI_HP0")} {return}
+  if {(($sys_zynq != 1 && $sys_zynq != 2) || ($p_cc && $sys_hpc0_interconnect_index != -1)) && ($p_name eq "sys_ps7/S_AXI_HP0")} {return}
   if {$sys_zynq == -1} {ad_mem_hpx_interconnect "SIM" $p_clk $p_name}
   if {$sys_zynq == 0} {ad_mem_hpx_interconnect "MEM" $p_clk $p_name}
   if {$sys_zynq == 1} {ad_mem_hpx_interconnect "HP0" $p_clk $p_name}
-  if {$sys_zynq == 2} {ad_mem_hpx_interconnect "HP0" $p_clk $p_name}
+  if {$sys_zynq == 2} {ad_mem_hpx_interconnect [expr {$p_cc ? "HPC0" : "HP0"}] $p_clk $p_name}
```

Comment:
I would revert adi_board.tcl, and have the user use ad_mem_hpc0/1_interconnect.
If he needs to dynamically use or not, he should at project/*/common/*_bd.tcl:
```
if {![info exists CACHE_COHERENCY]} {
  ad_mem_hp1_interconnect sys_dma_clk sys_ps8/S_AXI_HP1
} else {
  ad_mem_hpc0_interconnect sys_dma_clk sys_ps8/S_AXI_HPC0
}
```

But considering the projects that will use cache coherence are only with ps8, this snippet will be at project/*/zcu102/system_bd.tcl anyway, not requiring the if.

Reply:
This was one of the 2 possible ways to implement the interface selection based on the CACHE_COHERENCY parameter, using an if in _project_/common. I chose the adi_board.tcl approach to try to reduce the added changes in the common/_bd files and move them in the background, in adi_board.tcl. Though I admit that using an if in common allows for a more precise selection of the coherent interfaces.
Also, as a note, Versal platforms may also implement cache coherency in the future.